internal/generate/crd/crd.go

```diff
@@ -268,7 +280,31 @@ func (g crdGenerator) generateNonGo() (map[string][]byte, error) {
 	if err := checkCRDVersions(crd); err != nil {
 		return nil, fmt.Errorf("error checking CRD %s versions: %w", crd.GetName(), err)
 	}
-	b, err := k8sutil.GetObjectBytes(&crd, yaml.Marshal)
+
+	var (
+		b   []byte
+		err error
+	)
+	switch g.crdVersion {
+	case "v1beta1":
```

Comment:
Since we're going to switch to `v1.CustomResourceDefinition` by default in the future, we could set `crd := apiextv1.CustomResourceDefinition{}` above and return a `apiextv1.CustomResourceDefinition` from `newCRDForResource()`, then convert `v1 -> v1beta`. We'd also have to do some conversion if the file exists on disk and is `v1beta1`.

Reply:
Yeah, the reason I didn't do that was because of the possibility of a `v1beta1` CRD on disk. `v1` moved several top level fields from `v1beta1` into the `versions` items, so it is not possible to unmarshal the `v1beta1` YAML into a `v1` struct.

However, the `v1beta1` struct is a superset of the `v1` struct, so it is possible to unmarshal `v1` YAML into a `v1beta1` struct and then convert back.

Reply:
Is it possible to use the generic `apiextensions` struct for both?

Reply:
If I properly understood the suggestion here, IHMO is better we keep each version implementation since in the future we will just remove the v1beta1 and keep all generating v1 only.

Reply:
> Is it possible to use the generic apiextensions struct for both?

I can't get that to work either. For some reason (still trying to figure out why) everything gets marshaled without having the correct YAML layout (e.g. in the output there's actually an `objectMeta` key containing the GVK info, instead of that being inlined at the root).

Reply:
I’ve had trouble marshaling the generic CRD struct before. The current approach is fine then.

Reply:
:+1: 

Once we bump to v1.19, we'll have to basically revert this PR and change from v1beta1 to v1. This function will basically go back to how it looked before, except with the v1 types.